proofs/property_name.go

```diff
@@ -0,0 +1,94 @@
+package proofs
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Property struct {
+	Parent  *Property
```

Comment:
Could you use the same gofmt/golinting rules we use over here for this code: https://github.com/centrifuge/go-centrifuge/blob/develop/Makefile#L32

Reply:
I know this is still a WiP, adding it anyway just once as a reminder. The other thing that would be great is if you could add docstrings to all public interfaces.

Reply:
Should I be basing this on the develop branch?

Reply:
Oh sorry, that's another repo.

Reply:
Thanks! Feel free to copy the linter make command over into this repo :)